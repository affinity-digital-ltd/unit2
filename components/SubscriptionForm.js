import { Component } from 'react'

class SubscriptionForm extends Component {
  render () {
    return <div>
      <form onSubmit={this.props.emailSubscription}>
        <div className='input-group mb-3'>
          <input name='email' type='email' className='form-control' placeholder='Your email Address' required />
          <div className='input-group-append'>
            <input type='submit' value='Sign Up' className='btn btn-outline-secondary' />
          </div>
        </div>
      </form>
    </div>
  }
}

export default SubscriptionForm
