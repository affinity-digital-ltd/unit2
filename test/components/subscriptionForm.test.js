import SubscriptionForm from '../../components/SubscriptionForm'

describe('<SubscriptionForm />', () => {
  test('it renders correctly', () => {
    const component = shallow(
      <SubscriptionForm />
    )

    expect(component).toMatchSnapshot()
  })
})
