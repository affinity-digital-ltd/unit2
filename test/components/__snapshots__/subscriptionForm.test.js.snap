// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionForm /> it renders correctly 1`] = `
<div>
  <form>
    <div
      className="input-group mb-3"
    >
      <input
        className="form-control"
        name="email"
        placeholder="Your email Address"
        required={true}
        type="email"
      />
      <div
        className="input-group-append"
      >
        <input
          className="btn btn-outline-secondary"
          type="submit"
          value="Sign Up"
        />
      </div>
    </div>
  </form>
</div>
`;
